src/components/HubSpotForm.tsx

```diff
@@ -0,0 +1,323 @@
+
+declare global {
+    interface Window {
+        hbspt?: {
+            forms: {
+                create: ({
+                    region,
+                    portalId,
+                    formId,
+                    target,
+                    formInstanceId,
+                    onFormSubmit,
+                    onFormSubmitted,
+                    onFormReady,
+                }: HubSpotAPIProps) => CreateHubSpotFormProps
+            }
+        }
+        ChiliPiper?: ChiliPiperAPIProps
+        jQuery: () => void
+        ClearbitForHubspot: {
+            forms: []
+            addForm: () => void
+        }
+    }
+}
+
+interface HubSpotAPIProps {
+    region?: string
+    portalId: string
+    formId: string
+    target: string
+    formInstanceId?: string
+    onFormSubmit?: (object: { data: { name: string; value: string }[] }) => void
+    onFormReady?: ($form: CreateHubSpotFormProps) => void
+    onFormSubmitted?: () => void
+    inlineMessage?: string
+}
+
+interface CreateHubSpotFormProps {
+    [index: number]: HTMLFormElement
+    formId: string
+    onFormReady?: ($form: HTMLFormElement) => void
+    onFormSubmitted?: () => void
+    inlineMessage?: string
+    chiliPiper?: boolean
+}
+
+export interface HubSpotFormProps {
+    formId?: string
+    masterFormName?: 'contactMulti' | 'contactEmail' | 'gatedMulti' | 'gatedEmail'
+    onFormSubmitted?: () => void
+    inlineMessage?: string
+    chiliPiper?: boolean
+}
+
+interface ChiliPiperAPIProps {
+    submit: (
+        domain: string,
+        router: string,
+        options?: {
+            lead: { [key: string]: string | number }
+            handleSubmit?: boolean
+            formId?: string
+            debug?: boolean
+            map?: boolean
+            domain?: string
+            router?: string
+            title?: string
+            titleStyle?: string
+            onSuccess?: () => void
+            onError?: () => void
+            onClose?: () => void
+            onRouted?: () => void
+            closeOnOutside?: boolean
+            dynamicRedirectLink?: string
+            mobileRedirectLink?: string
+            injectRootCss?: boolean
+            locale?: string
+            webHook?: string
+        }
+    ) => void
+}
+
+interface HubSpotEventProps {
+    type: string
+    eventName: string
+    data: { name: string; value: string }[]
+}
+
+/**
+ * These are our Master Forms that are used throughout our codebase. Each
+ * masterFormName is used as an identifier to map to a specific master form id.
+ */
+const masterForms: { [key: string]: string } = {
+    // Contact Us Email Only Form
+    contactEmail: 'a26c29e7-cd79-429d-a2ac-43f694734c46',
+
+    // Contact Us Multi Field Form
+    contactMulti: 'e090296f-84f5-4bcb-9093-a533336841b4',
+
+    // Gated Content Email Only Form
+    gatedEmail: '9b2539ad-feaa-4dd2-b6b4-2439c5bc98da',
+
+    // Gated Content Multi Field Form
+    gatedMulti: '1fb4ef6c-f233-48ba-9f43-a88f19528282',
+}
```

Comment:
Thanks @st0nebraker and thanks for testing! Yes the `anonymous_user_id` and `form_submission_source` are both hidden form fields. You should be able to see that data in a form submission on HubSpot under submissions of the form you submitted.

I'll have to look into why the HubSpot form shows blank when navigating back and forth in page history.

I think the enum is probably trivial in this case since we limit the `masterFormName` on `HubSpotProps` with a union of the object dictionary keys.

Reply:
Interesting- I'm not seeing `anonymous_user_id` or `form_submission_source` on a new form I tested. But I see `form_submission_source` and `first_source_url` on some random 
 forms. What are you seeing? We can wait until after your time off to handle this btw :)
<img width="1055" alt="image" src="https://user-images.githubusercontent.com/59381432/178081967-07baad23-ee1e-4e28-87ec-7e549413973a.png">
<img width="1366" alt="image" src="https://user-images.githubusercontent.com/59381432/178082117-f9cbe187-64e3-4626-9fe7-ab14c21f567b.png">

Reply:
Thanks for noticing this @st0nebraker! I just pushed up a fix which populates the hidden form fields properly now. This should be working now. Can you re-test a form submission?

Reply:
Huzzah! I'm seeing all the hidden fields now for submissions 🥳 
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/59381432/179819621-abc21da6-5cc3-496d-9963-e479e55a3008.png">